docs/messaging/nats-integration.md

```diff
@@ -2,110 +2,201 @@
+
+> [!TIP]
+> If you'd rather connect to an existing NATS server, call <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> instead. For more information, see [Reference existing resources](../fundamentals/app-host-overview.md#reference-existing-resources).
+
+### Add NATS server resource with JetStream
+
+To add the [NATS JetStream](https://docs.nats.io/nats-concepts/jetstream) to the NATS server resource, call the <xref:Aspire.Hosting.NatsBuilderExtensions.WithJetStream*> method:
 
 ```csharp
-public class ExampleService(INatsConnection client)
-{
-    // Use client...
-}
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+                  .WithJetStream();
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
+
+// After adding all resources, run the app...
 ```
 
-## App host usage
+The NATS JetStream functionality provides a built-in persistence engine called JetStream which enables messages to be stored and replayed at a later time. You can optionally provide a `srcMountPath` parameter to specify the path to the JetStream data directory on the host machine (the provided mount path maps to the container's `-sd` argument).
+
+### Add NATS server resource with data volume
+
+To add a data volume to the NATS server resource, call the <xref:Aspire.Hosting.NatsBuilderExtensions.WithDataVolume*> method on the NATS server resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+                  .WithDataVolume(isReadOnly: false);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
 
-To model the Nats resource in the app host, install the [Aspire.Hosting.Nats](https://www.nuget.org/packages/Aspire.Hosting.Nats) NuGet package in the [app host](xref:aspire/app-host) project.
+// After adding all resources, run the app...
+```
+
+The data volume is used to persist the NATS server data outside the lifecycle of its container. The data volume is mounted at the `/var/lib/nats` path in the NATS server container and when a `name` parameter isn't provided, the name is generated at random. For more information on data volumes and details on why they're preferred over [bind mounts](#add-nats-server-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
+
+### Add NATS server resource with data bind mount
+
+To add a data bind mount to the NATS server resource, call the <xref:Aspire.Hosting.NatsBuilderExtensions.WithDataBindMount*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+                  .WithDataBindMount(
+                      source: "/NATS/Data",
+                      isReadOnly: false);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
+
+// After adding all resources, run the app...
+```
+
```

Comment:
Is this a good place to insert a TIP or WARNING about why they should use a data volume instead?

Reply:
Done